src/core/trulens/core/utils/containers.py

```diff
@@ -134,6 +136,13 @@ def pop(self, blocking: bool = True) -> Optional[T]:
             if len(self.content) == 0:
                 self.event_nonempty.clear()
 
+            self.read_lock.release()
+
+        else:
+            # Acquire was used with blocking=False and the lock would have
+            # blocked.
```

Comment:
shouldn't this read sth like
```Acquire was used with blocking=False and should return immediately if the lock is not yet released```? 

or maybe I'm misunderstanding the comment here

Reply:
I think we are both saying the same thing. Added some more text to maybe unify.